pkgs/development/tools/mmixware/default.nix

```diff
@@ -0,0 +1,42 @@
+{ stdenv, fetchurl, tetex }:
+
+stdenv.mkDerivation {
+  name = "mmixware";
+
+  src = fetchurl {
+    url = "https://www-cs-faculty.stanford.edu/~knuth/programs/mmix.tar.gz";
+    sha256 = "16645rmykq0b5wgpjzlfi1ap1j6ig88lra2b9bqfsrrqlawajpp9";
+  };
+  unpackPhase = ''
+    mkdir src
+    unpackFile $src src
+  '';
+  srcRoot = "src";
+
+  hardeningDisable = [ "format" ];
+
+  patchPhase = ''
+    substituteInPlace Makefile --replace 'rm abstime.h' ""
+  '';
+
+  nativeBuildInputs = [ tetex ];
+  enableParallelBuilding = true;
+
+  makeFlags = [ "all" "doc" ];
+
+  installPhase = ''
+    mkdir -p $out/share/doc
```

Comment:
When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`.